src/ast.ts

```diff
@@ -302,31 +302,12 @@ function joinShouldIncludeCompilations(...compilations: string[]) {
   // remove empty strings
   let filteredCompilations = compilations.filter(it => it);
 
-  // FIXME(boopathi): Maybe, find a better fix?
-  //
-  // remove "true" since we are joining with '&&' as `true && X` = `X`
-  // This prevents an explosion of `&& true` which could break
-  // V8's internal size limit for string.
-  //
-  // Note: the `true` appears if a field does not have a skip/include directive
-  // So, the more nested the query is, the more of unnecessary `&& true`
-  // we get.
-  //
-  // Failing to do this results in [RangeError: invalid array length]
-  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Invalid_array_length
```

Comment:
It's still valuable to retain this comment and update it. The FIXME is no longer required. But the explanation of why deduplication exists would help.